admin-console/src/main/java/com/agiletec/apsadmin/portal/specialwidget/navigator/NavigatorWidgetConfigAction.java

```diff
@@ -219,9 +219,13 @@ protected void createExpressions(List<NavigatorExpression> expressions) {
 
 	private void addPages(IPage page, List<IPage> pages) {
 		pages.add(page);
-		IPage[] children = page.getChildren();
+		boolean isOnline = page.isOnline();
+		String[] children = page.getChildrenCodes();
 		for (int i = 0; i < children.length; i++) {
-			this.addPages(children[i], pages);
+			IPage child = (isOnline)
+					? this.getPageManager().getOnlinePage(children[i])
+					: this.getPageManager().getDraftPage(children[i]);
+			this.addPages(child, pages);

```

Comment:
See above. Move to a Utility.

Reply:
See the reply for the class PageAction.java